Segmentation fault (core dumped) while running tests

Hello,

Whilst trying to prep a release for `20210728.00` in Ubuntu, the package failed to build. More specifically, the tests crashed with a segfault. Here are the logs:

```
[...]
 debian/rules build
dh build
   dh_update_autotools_config
   dh_autoreconf
   debian/rules override_dh_auto_configure
make[1]: Entering directory '/<<PKGBUILDDIR>>'
cp -r /usr/src/gtest/ ./
dh_auto_configure
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
   dh_auto_build
	make -j8 "INSTALL=install --strip-program=true"
make[1]: Entering directory '/<<PKGBUILDDIR>>'
make -C src all
make[2]: Entering directory '/<<PKGBUILDDIR>>/src'
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c  -c -o nss/nss_oslogin.o nss/nss_oslogin.cc
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c  -c -o oslogin_utils.o oslogin_utils.cc
cc -fPIC -Wall -g -Wstrict-prototypes -Iinclude -I/usr/include/json-c  -c -o nss/nss_cache_oslogin.o nss/nss_cache_oslogin.c
cc -fPIC -Wall -g -Wstrict-prototypes -Iinclude -I/usr/include/json-c  -c -o nss/compat/getpwent_r.o nss/compat/getpwent_r.c
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c  -c -o pam/pam_oslogin_login.o pam/pam_oslogin_login.cc
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c  -c -o pam/pam_oslogin_admin.o pam/pam_oslogin_admin.cc
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c  -c -o cache_refresh/cache_refresh.o cache_refresh/cache_refresh.cc
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c  -c -o authorized_keys/authorized_keys.o authorized_keys/authorized_keys.cc
cache_refresh/cache_refresh.cc: In function ‘int refreshgroupcache()’:
cache_refresh/cache_refresh.cc:171:35: warning: catching polymorphic type ‘class std::ios_base::failure’ by value [-Wcatch-value=]
  171 |     catch (std::ofstream::failure e) {
      |                                   ^
cache_refresh/cache_refresh.cc:180:33: warning: catching polymorphic type ‘class std::ios_base::failure’ by value [-Wcatch-value=]
  180 |   catch (std::ofstream::failure e) {
      |                                 ^
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c -shared -Wl,-z,defs -Wl,-soname,libnss_oslogin.so.2 nss/nss_oslogin.o oslogin_utils.o -o libnss_oslogin-20210728.00.so -lcurl -ljson-c
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c -shared -Wl,-z,defs -Wl,-soname,libnss_cache_oslogin.so.2 nss/nss_cache_oslogin.o nss/compat/getpwent_r.o oslogin_utils.o -o libnss_cache_oslogin-20210728.00.so -lcurl -ljson-c
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c -shared pam/pam_oslogin_login.o oslogin_utils.o -o pam_oslogin_login.so -lpam -lcurl -ljson-c
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c -shared pam/pam_oslogin_admin.o oslogin_utils.o -o pam_oslogin_admin.so -lpam -lcurl -ljson-c
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c cache_refresh/cache_refresh.o oslogin_utils.o -o google_oslogin_nss_cache -lcurl -ljson-c
g++ -fPIC -Wall -g -Iinclude -I/usr/include/json-c authorized_keys/authorized_keys.o oslogin_utils.o -o google_authorized_keys -lcurl -ljson-c
make[2]: Leaving directory '/<<PKGBUILDDIR>>/src'
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
   debian/rules override_dh_auto_test
make[1]: Entering directory '/<<PKGBUILDDIR>>'
(cd gtest && cmake . && /usr/bin/make )
-- The CXX compiler identification is GNU 11.2.0
-- The C compiler identification is GNU 11.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE) 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /<<PKGBUILDDIR>>/gtest
make[2]: Entering directory '/<<PKGBUILDDIR>>/gtest'
make[3]: Entering directory '/<<PKGBUILDDIR>>/gtest'
make[4]: Entering directory '/<<PKGBUILDDIR>>/gtest'
Scanning dependencies of target gtest
make[4]: Leaving directory '/<<PKGBUILDDIR>>/gtest'
make[4]: Entering directory '/<<PKGBUILDDIR>>/gtest'
[ 25%] Building CXX object CMakeFiles/gtest.dir/src/gtest-all.cc.o
[ 50%] Linking CXX static library lib/libgtest.a
make[4]: Leaving directory '/<<PKGBUILDDIR>>/gtest'
[ 50%] Built target gtest
make[4]: Entering directory '/<<PKGBUILDDIR>>/gtest'
Scanning dependencies of target gtest_main
make[4]: Leaving directory '/<<PKGBUILDDIR>>/gtest'
make[4]: Entering directory '/<<PKGBUILDDIR>>/gtest'
[ 75%] Building CXX object CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[100%] Linking CXX static library lib/libgtest_main.a
make[4]: Leaving directory '/<<PKGBUILDDIR>>/gtest'
[100%] Built target gtest_main
make[3]: Leaving directory '/<<PKGBUILDDIR>>/gtest'
make[2]: Leaving directory '/<<PKGBUILDDIR>>/gtest'
make -C test CXXFLAGS="-g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -I/<<PKGBUILDDIR>>/gtest" GTEST_DIR=/<<PKGBUILDDIR>>/gtest
make[2]: Entering directory '/<<PKGBUILDDIR>>/test'
g++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -I/<<PKGBUILDDIR>>/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/<<PKGBUILDDIR>>/src/include -I/usr/include/json-c -I/<<PKGBUILDDIR>>/gtest -isystem /<<PKGBUILDDIR>>/gtest/include  -c -o oslogin_utils_test.o oslogin_utils_test.cc
g++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -I/<<PKGBUILDDIR>>/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/<<PKGBUILDDIR>>/src/include -I/usr/include/json-c -I/<<PKGBUILDDIR>>/gtest -isystem /<<PKGBUILDDIR>>/gtest/include -c /<<PKGBUILDDIR>>/gtest/src/gtest-all.cc
g++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -I/<<PKGBUILDDIR>>/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/<<PKGBUILDDIR>>/src/include -I/usr/include/json-c -I/<<PKGBUILDDIR>>/gtest -isystem /<<PKGBUILDDIR>>/gtest/include oslogin_utils_test.o /<<PKGBUILDDIR>>/src/oslogin_utils.o gtest-all.o -o test_runner -lcurl -ljson-c -lpthread
g++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -I/<<PKGBUILDDIR>>/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/<<PKGBUILDDIR>>/src/include -I/usr/include/json-c -I/<<PKGBUILDDIR>>/gtest -isystem /<<PKGBUILDDIR>>/gtest/include  -c -o oslogin_test.o oslogin_test.cc
g++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -I/<<PKGBUILDDIR>>/gtest -Wdate-time -D_FORTIFY_SOURCE=2 -I/<<PKGBUILDDIR>>/src/include -I/usr/include/json-c -I/<<PKGBUILDDIR>>/gtest -isystem /<<PKGBUILDDIR>>/gtest/include oslogin_test.o gtest-all.o -o new_test_runner -lcurl -ljson-c -lpthread
./test_runner --gtest_output=xml --gtest_filter=*-GetGroupByTest.*:GetUsersForGroupTest.*
Note: Google Test filter = *-GetGroupByTest.*:GetUsersForGroupTest.*
[==========] Running 35 tests from 12 test suites.
[----------] Global test environment set-up.
[----------] 2 tests from BufferManagerTest
[ RUN      ] BufferManagerTest.TestAppendString
[       OK ] BufferManagerTest.TestAppendString (0 ms)
[ RUN      ] BufferManagerTest.TestAppendStringTooLarge
[       OK ] BufferManagerTest.TestAppendStringTooLarge (0 ms)
[----------] 2 tests from BufferManagerTest (0 ms total)

[----------] 4 tests from NssCacheTest
[ RUN      ] NssCacheTest.TestLoadJsonArray
[       OK ] NssCacheTest.TestLoadJsonArray (1 ms)
[ RUN      ] NssCacheTest.TestLoadJsonPartialArray
[       OK ] NssCacheTest.TestLoadJsonPartialArray (0 ms)
[ RUN      ] NssCacheTest.TestLoadJsonFinalResponse
[       OK ] NssCacheTest.TestLoadJsonFinalResponse (0 ms)
[ RUN      ] NssCacheTest.ResetNullPtrTest
[       OK ] NssCacheTest.ResetNullPtrTest (0 ms)
[----------] 4 tests from NssCacheTest (1 ms total)

[----------] 8 tests from ParseJsonPasswdTest
[ RUN      ] ParseJsonPasswdTest.ParseJsonToPasswdSucceeds
[       OK ] ParseJsonPasswdTest.ParseJsonToPasswdSucceeds (0 ms)
[ RUN      ] ParseJsonPasswdTest.ParseJsonToPasswdSucceedsWithHighUid
[       OK ] ParseJsonPasswdTest.ParseJsonToPasswdSucceedsWithHighUid (0 ms)
[ RUN      ] ParseJsonPasswdTest.ParseJsonToPasswdSucceedsWithStringUid
[       OK ] ParseJsonPasswdTest.ParseJsonToPasswdSucceedsWithStringUid (0 ms)
[ RUN      ] ParseJsonPasswdTest.ParseJsonToPasswdNoLoginProfilesSucceeds
[       OK ] ParseJsonPasswdTest.ParseJsonToPasswdNoLoginProfilesSucceeds (0 ms)
[ RUN      ] ParseJsonPasswdTest.ParseJsonToPasswdFailsWithERANGE
[       OK ] ParseJsonPasswdTest.ParseJsonToPasswdFailsWithERANGE (0 ms)
[ RUN      ] ParseJsonPasswdTest.ParseJsonToPasswdFailsWithEINVAL
[       OK ] ParseJsonPasswdTest.ParseJsonToPasswdFailsWithEINVAL (0 ms)
[ RUN      ] ParseJsonPasswdTest.ValidatePartialJsonResponse
[       OK ] ParseJsonPasswdTest.ValidatePartialJsonResponse (0 ms)
[ RUN      ] ParseJsonPasswdTest.ValidateInvalidJsonResponse
[       OK ] ParseJsonPasswdTest.ValidateInvalidJsonResponse (0 ms)
[----------] 8 tests from ParseJsonPasswdTest (0 ms total)

[----------] 5 tests from ParseJsonToGroupsTest
[ RUN      ] ParseJsonToGroupsTest.ParseJsonToGroupsSucceeds
[       OK ] ParseJsonToGroupsTest.ParseJsonToGroupsSucceeds (0 ms)
[ RUN      ] ParseJsonToGroupsTest.ParseJsonToGroupsSucceedsWithHighGid
[       OK ] ParseJsonToGroupsTest.ParseJsonToGroupsSucceedsWithHighGid (0 ms)
[ RUN      ] ParseJsonToGroupsTest.ParseJsonToGroupsSucceedsWithStringGid
[       OK ] ParseJsonToGroupsTest.ParseJsonToGroupsSucceedsWithStringGid (0 ms)
[ RUN      ] ParseJsonToGroupsTest.ParseJsonToGroupsFails
[       OK ] ParseJsonToGroupsTest.ParseJsonToGroupsFails (0 ms)
[ RUN      ] ParseJsonToGroupsTest.TestGroups
[       OK ] ParseJsonToGroupsTest.TestGroups (0 ms)
[----------] 5 tests from ParseJsonToGroupsTest (0 ms total)

[----------] 2 tests from ParseJsonToUsersTest
[ RUN      ] ParseJsonToUsersTest.ParseJsonToUsersSucceeds
[       OK ] ParseJsonToUsersTest.ParseJsonToUsersSucceeds (0 ms)
[ RUN      ] ParseJsonToUsersTest.ParseJsonToUsersEmptyGroupSucceeds
[       OK ] ParseJsonToUsersTest.ParseJsonToUsersEmptyGroupSucceeds (0 ms)
[----------] 2 tests from ParseJsonToUsersTest (0 ms total)

[----------] 2 tests from ParseJsonEmailTest
[ RUN      ] ParseJsonEmailTest.SuccessfullyParsesEmail
[       OK ] ParseJsonEmailTest.SuccessfullyParsesEmail (0 ms)
[ RUN      ] ParseJsonEmailTest.FailsParseEmail
[       OK ] ParseJsonEmailTest.FailsParseEmail (0 ms)
[----------] 2 tests from ParseJsonEmailTest (0 ms total)

[----------] 4 tests from ParseJsonSshKeyTest
[ RUN      ] ParseJsonSshKeyTest.ParseJsonToSshKeysSucceeds
[       OK ] ParseJsonSshKeyTest.ParseJsonToSshKeysSucceeds (0 ms)
[ RUN      ] ParseJsonSshKeyTest.ParseJsonToSshKeysMultipleKeys
[       OK ] ParseJsonSshKeyTest.ParseJsonToSshKeysMultipleKeys (0 ms)
[ RUN      ] ParseJsonSshKeyTest.ParseJsonToSshKeysFiltersExpiredKeys
[       OK ] ParseJsonSshKeyTest.ParseJsonToSshKeysFiltersExpiredKeys (0 ms)
[ RUN      ] ParseJsonSshKeyTest.ParseJsonToSshKeysFiltersMalformedExpiration
[       OK ] ParseJsonSshKeyTest.ParseJsonToSshKeysFiltersMalformedExpiration (0 ms)
[----------] 4 tests from ParseJsonSshKeyTest (0 ms total)

[----------] 1 test from ParseJsonAuthorizeSuccess
[ RUN      ] ParseJsonAuthorizeSuccess.SuccessfullyAuthorized
[       OK ] ParseJsonAuthorizeSuccess.SuccessfullyAuthorized (0 ms)
[----------] 1 test from ParseJsonAuthorizeSuccess (0 ms total)

[----------] 2 tests from ValidateUserNameTest
[ RUN      ] ValidateUserNameTest.ValidateValidUserNames
[       OK ] ValidateUserNameTest.ValidateValidUserNames (4 ms)
[ RUN      ] ValidateUserNameTest.ValidateInvalidUserNames
[       OK ] ValidateUserNameTest.ValidateInvalidUserNames (4 ms)
[----------] 2 tests from ValidateUserNameTest (8 ms total)

[----------] 2 tests from ParseJsonKeyTest
[ RUN      ] ParseJsonKeyTest.TestKey
[       OK ] ParseJsonKeyTest.TestKey (0 ms)
[ RUN      ] ParseJsonKeyTest.TestMissingKey
[       OK ] ParseJsonKeyTest.TestMissingKey (0 ms)
[----------] 2 tests from ParseJsonKeyTest (0 ms total)

[----------] 2 tests from ParseJsonChallengesTest
[ RUN      ] ParseJsonChallengesTest.TestChallenges
Segmentation fault (core dumped)
make[2]: *** [Makefile:31: non_network_tests] Error 139
make[2]: Leaving directory '/<<PKGBUILDDIR>>/test'
make[1]: *** [debian/rules:16: override_dh_auto_test] Error 2
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:8: build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
--------------------------------------------------------------------------------
```

Any idea what's up? Thanks in advance! \o/

--- a/src/oslogin_utils.cc
+++ b/src/oslogin_utils.cc
@@ -871,12 +871,10 @@ bool ParseJsonToKey(const string& json, const string& key, string* response) {
 
 
   if (!json_object_object_get_ex(root, key.c_str(), &json_response)) {
-    json_object_put(root);
     goto cleanup;
   }
 
   if (!(c_response = json_object_get_string(json_response))) {
-    json_object_put(root);
     goto cleanup;
   }
 

